### ISSUE TITLE
Implement "chain comparison" annotator.

### ISSUE THREAD

**Issue by matklad** [intellij-rust/intellij-rust#489]
Problem: some users (me included) often times forget to place `::` before generics in expression context. That is, they write `Vec::new<i32>()` instead of `Vec::new::<i32>()`

Solution: this problem is neatly solved in Rust. Chained comparisons like `1 < 2 < 3` are forbidden on the syntactic level (in contrast to other languages, where they are ruled out during type checking), so the parser can provide a meaningful "did you mean `::<>`" error. For the plugin, I thing the best way to handle this is to add the case to the [InvalidSyntaxAnnotator](https://github.com/intellij-rust/intellij-rust/blob/cdf2cb4830c0cbb4df2ac4c406f267c9bd23c86b/src/main/kotlin/org/rust/ide/annotator/RustInvalidSyntaxAnnotator.kt).

The plan:
- Make sure that chained comparisons are indeed always forbidden in Rust and collect a bunch of test cases. 
- Add a new test and new pattern to the annotator. 
- (optional, separate PR) add a quick fix to turn `foo<T>()` into `foo::<T>()`.

**Comment by mkaput** [comment]
Some real-life test case (should be annotated as error, but is not):
```rust
fn main() {
    let lit = b
    <caret>"Hello, World";
}
```

Generated PSI (as of 2016-08-07):
```
FILE(0,57)
  RustFnItemElementImpl(FN_ITEM)(0,56)
    PsiElement(fn)('fn')(0,2)
    PsiWhiteSpace(' ')(2,3)
    PsiElement(<IDENTIFIER>)('main')(3,7)
    RustParametersElementImpl(PARAMETERS)(7,9)
      PsiElement(()('(')(7,8)
      PsiElement())(')')(8,9)
    PsiWhiteSpace(' ')(9,10)
    RustBlockElementImpl(BLOCK)(10,56)
      PsiElement({)('{')(10,11)
      PsiWhiteSpace('\n    ')(11,16)
      RustLetDeclElementImpl(LET_DECL)(16,54)
        PsiElement(let)('let')(16,19)
        PsiWhiteSpace(' ')(19,20)
        RustPatIdentElementImpl(PAT_IDENT)(20,23)
          RustPatBindingElementImpl(PAT_BINDING)(20,23)
            PsiElement(<IDENTIFIER>)('lit')(20,23)
        PsiWhiteSpace(' ')(23,24)
        PsiElement(=)('=')(24,25)
        PsiWhiteSpace(' ')(25,26)
        RustBinaryExprElementImpl(BINARY_EXPR)(26,53)
          RustBinaryExprElementImpl(BINARY_EXPR)(26,38)
            RustPathExprElementImpl(PATH_EXPR)(26,27)
              RustPathElementImpl(PATH)(26,27)
                PsiElement(<IDENTIFIER>)('b')(26,27)
            PsiWhiteSpace('\n    ')(27,32)
            PsiElement(<)('<')(32,33)
            RustPathExprElementImpl(PATH_EXPR)(33,38)
              RustPathElementImpl(PATH)(33,38)
                PsiElement(<IDENTIFIER>)('caret')(33,38)
          PsiElement(>)('>')(38,39)
          RustLitExprElementImpl(LIT_EXPR)(39,53)
            RustStringLiteralImpl(<STRING>)('"Hello, World"')(39,53)
        PsiElement(;)(';')(53,54)
      PsiWhiteSpace('\n')(54,55)
      PsiElement(})('}')(55,56)
  PsiWhiteSpace('\n')(56,57)
```

**Comment by allengeorge** [comment]
I'll tackle this.

**Comment by la10736** [comment]
@matklad  I would like to start to work on this. Are there some others tests that check rust's compiler features like the ones that we should write for _chain comparison_? If yes where are they located and where is better to puts these ones? 

Thanks

**Comment by matklad** [comment]
@la10736 take a look at the [InvalidSyntaxAnnotatorTest](https://github.com/intellij-rust/intellij-rust/blob/master/src/test/kotlin/org/rust/ide/annotator/RustInvalidSyntaxAnnotatorTest.kt). 

Usually invalid syntax is handled directly in the grammar, but sometimes it make sense to parse something invalid and annotate it as syntax error later. This is precisely the job of the invalid syntax annotator.

**Comment by mkaput** [comment]
Solved by #945
